contracts/test/RefinanceTest.t.sol

```diff
@@ -0,0 +1,274 @@
+        /************************************/
+
+        pool.claim(address(loanV2), address(debtLockerFactory));
+
+
+        /*****************/
+        /*** Refinance ***/
+        /*****************/
+
+        // Refinance parameters
+
+        uint256[3] memory refinanceTermDetails = [
+            uint256(15 days),  // 15 day grace period
+            uint256(45 days),  // 45 day payment interval
+            uint256(3)
+        ];
+
+        // 5 BTC @ ~$58k = $290k = 29% collateralized, interest only
+        uint256[3] memory refinanceRequests = [uint256(2 * BTC), uint256(2_000_000 * USD), uint256(2_000_000 * USD)];  
+
+        uint256[4] memory refinanceRates = [uint256(0.10e18), uint256(0), uint256(0.04e18), uint256(0.4e18)];
+
+        uint256 principalIncrease = refinanceRequests[1] - requests[1];
+
+        bytes[] memory data = new bytes[](9);
+        data[0] = abi.encodeWithSelector(Refinancer.setGracePeriod.selector,       refinanceTermDetails[0]);
+        data[1] = abi.encodeWithSelector(Refinancer.setPaymentInterval.selector,   refinanceTermDetails[1]);
+        data[2] = abi.encodeWithSelector(Refinancer.setPaymentsRemaining.selector, refinanceTermDetails[2]);
+
+        data[3] = abi.encodeWithSelector(Refinancer.setCollateralRequired.selector, refinanceRequests[0]);
+        data[4] = abi.encodeWithSelector(Refinancer.increasePrincipal.selector,     principalIncrease);
+        data[5] = abi.encodeWithSelector(Refinancer.setEndingPrincipal.selector,    refinanceRequests[2]);
+
+        data[6] = abi.encodeWithSelector(Refinancer.setInterestRate.selector,        refinanceRates[0]);
+        data[7] = abi.encodeWithSelector(Refinancer.setLateFeeRate.selector,         refinanceRates[2]);
+        data[8] = abi.encodeWithSelector(Refinancer.setLateInterestPremium.selector, refinanceRates[3]);
+
+        borrower.loan_proposeNewTerms(address(loanV2), address(refinancer), data);
+
+        // Pool Delegate funds loan again for increasing amount
+        pool.fundLoan(address(loanV2), address(debtLockerFactory), principalIncrease);
+
+        // Pool Delegate(address this) accepts new terms on Debt Locker
+        DebtLocker debtLocker = DebtLocker(loanV2.lender());
+
+        debtLocker.acceptNewTerms(address(refinancer), data, principalIncrease);
+        
+        assertEq(loanV2.gracePeriod(),       refinanceTermDetails[0]);
+        assertEq(loanV2.paymentInterval(),   refinanceTermDetails[1]);
+        assertEq(loanV2.paymentsRemaining(), refinanceTermDetails[2]);
+
+        assertEq(loanV2.principalRequested(), requests[1] + principalIncrease);
+        assertEq(loanV2.principal(),          requests[1] + principalIncrease);
+        assertEq(loanV2.collateralRequired(), refinanceRequests[0]);
+
+        assertEq(loanV2.interestRate(),        refinanceRates[0]);
+        assertEq(loanV2.lateFeeRate(),         refinanceRates[2]);
+        assertEq(loanV2.lateInterestPremium(), refinanceRates[3]);
+
```

Comment:
Lets do another payment and claim after the refinance and explictly assert the values to show a change in the amount

Reply:
Lets also set endingPrincipal to be lower than 2m to demonstrate that we can move from interest-only to amortized using a refiannce